### PR TITLE
Pickle 5

### LINKED ISSUE #3635: New arguments for pickle 5
Python 3.8.1, mypy 0.770+dev.861f01c3be99b9faf2e8d42e316aee4ec6e571c4

```python
pickle.dumps(obj, protocol=5, buffer_callback=f)
```
Output:
```
error: Unexpected keyword argument "buffer_callback" for "dumps"
```

### PR DESCRIPTION
**Pull request by crusaderky**
Closes #3635

### CODE REVIEW

**Comment by crusaderky** [comment]
The suite 'mypy self test' is failing with a very uninformative message ``OSError: cannot send to <Channel id=1 closed>``; help would be appreciated

https://travis-ci.org/python/typeshed/jobs/639597167?utm_medium=notification&utm_source=github_status

**Comment by hauntsaninja** [comment]
Try forcing another CI build? Things seem to be working now, eg, for #3634

**Comment by srittau** [comment]
I restarted CI.

**Comment by crusaderky** [comment]
Thank you - all green now. Ready for review.

File: stdlib/2and3/pickle.pyi [hunk 1]
```diff
@@ -1,20 +1,46 @@
...
     DEFAULT_PROTOCOL: int
 
 
-if sys.version_info >= (3, 0):
+if sys.version_info >= (3, 8):
+    # TODO: holistic design for buffer interface (typing.Buffer?)
+
+    class PickleBuffer:
+        # buffer must be a buffer-providing object
+        def __init__(self, buffer: Any): ...
```
Comment (srittau) [inline stdlib/2and3/pickle.pyi]:
```suggestion
        def __init__(self, buffer: Any) -> None: ...
```

File: stdlib/2and3/pickle.pyi [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
+        def raw(self) -> memoryview: ...
+        def release(self) -> None: ...
+    def dump(
+        obj: Any, file: IO[bytes], protocol: Optional[int] = ..., *,
+        fix_imports: bool = ...,
+        buffer_callback: Optional[Callable[[PickleBuffer], None]] = ...
```
Comment (srittau) [inline stdlib/2and3/pickle.pyi]: If I understand the documentation correctly, the callback can return a truish value to signal in-band data. I think we should just use `Any` as return type for the callback. Also affects other methods below. I'd also suggest to define a type alias for the callback type (e.g. `_BufferCallback`) and use it throughout the file.

**Comment by crusaderky** [comment]
All comments implemented, with the exception of the () default for buffers where I feel the cpython implementation should be corrected

**Comment by crusaderky** [comment]
@hauntsaninja I opened https://bugs.python.org/issue39435

**Pull request merged**